python/ray/dag/compiled_dag_node.py

```diff
@@ -75,6 +76,9 @@ def do_exec_tasks(
     Args:
         tasks: the executable tasks corresponding to the actor methods.
     """
+    ctx = ChannelContext.get_current()
+    self._serialization_ctx = ctx.serialization_context
```

Comment:
Yes, we want to avoid modifying the base ray Actor class, so instead of defining the method `do_exec_tasks` on the base Actor class, we instead pass it to the generic `__ray_call__` method, which takes in a lambda and the lambda args/kwargs.